tests/unittests/CoreGraphics/CGGeometryTests.mm

```diff
@@ -52,4 +52,122 @@
     CGSize inverseTransformedSize = CGSizeApplyAffineTransform(transformedSize, CGAffineTransformInvert(rotate));
     EXPECT_NEAR(size.width, inverseTransformedSize.width, 0.001);
     EXPECT_NEAR(size.height, inverseTransformedSize.height, 0.001);
-}
\ No newline at end of file
+}
+
+class CGGeometryDivRect : public ::testing::TestWithParam<::testing::tuple<CGRect, CGFloat, CGRectEdge, CGRect, CGRect>> {};
+
+TEST_P(CGGeometryDivRect, RectDivide) {
+    CGRect rect = ::testing::get<0>(GetParam());
+    CGFloat amount = ::testing::get<1>(GetParam());
+    CGRectEdge edge = ::testing::get<2>(GetParam());
+    CGRect rSlice = ::testing::get<3>(GetParam());
+    CGRect rRemainder = ::testing::get<4>(GetParam());
+
+    CGRect slice, remainder;
+    CGRectDivide(rect, &slice, &remainder, amount, edge);
+
+    EXPECT_TRUE(CGRectEqualToRect(rSlice, slice));
+    EXPECT_TRUE(CGRectEqualToRect(rRemainder, remainder));
+}
+
+std::tr1::tuple<CGRect, CGFloat, CGRectEdge, /*slice*/ CGRect, /*remainder*/ CGRect> const cCGRectDivideResults[] = {
+
+    ::testing::make_tuple(CGRectMake(0, 0, 100, 100),
```

Comment:
The point of comments is to clarify difficult-to-read sections of code. Since this is a pile of constant values in no discernible order, it is a difficult-to-read section of code.

Adding comments will only serve to increase its clarity.

It also helps us think about all the test cases and whether there are gaps.